src/ape/api/explorers.py

```diff
@@ -36,3 +39,15 @@ def get_transaction_url(self, transaction_hash: str) -> str:
         Returns:
             str
         """
+
+    @abstractmethod
+    def get_contract_type(self, address: AddressType) -> Optional[ContractType]:
+        """
+        Get the contract type for a given address if it has been published in explorer
+
+        Args:
+            address (str): The contract address.
+
+        Returns:
+            ContractType if published, else None
```

Comment:
```suggestion
            :class:`~ape.contracts.ContractType` if published, else ``None``.
```

Reply:
@unparalleled-js we had a question I didn't know offhand, is it acceptable docstring syntax to write a return this way? Thought it was more useful that just saying "returns `Optional[ContractType]` which really doesn't tell you as much about the context shift